### ISSUE TITLE
NELL dataset doesn't have num_features and num_node_features

### ISSUE THREAD

**Issue by gau-nernst** [pyg-team/pytorch_geometric#5053]
### 🐛 Describe the bug

num_features and num_node_features are 0 for NELL dataset. I'm not familiar with PyG dataset internals, but I suspect it is due to node features in NELL is a `SparseTensor` instance instead of a normal `Tensor`.

```python
from torch_geometric.datasets import NELL

ds = NELL("data")
print(ds.num_node_features)  # 0
print(ds.num_features)       # 0
```

### Environment

* PyG version: 2.0.4
* PyTorch version: 1.11.0
* OS: macOS Monterey 12.4
* Python version: 3.10.5
* CUDA/cuDNN version: NA
* How you installed PyTorch and PyG (`conda`, `pip`, source): PyTorch through conda, PyG through pip
* Any other relevant information (*e.g.*, version of `torch-scatter`):

**Comment by rusty1s** [comment]
Thanks for reporting. Let me take a look.

**Comment by rusty1s** [comment]
Fixed in #5089! Thanks for reporting :)
